src/examples/design/icons/SvgSearch.tsx

```diff
@@ -29,85 +31,148 @@ const StyledSvgWrapper = styled.div<{ isAnswerBot?: boolean }>`
   color: ${p => p.isAnswerBot && '#d6eef1'};
 `;
 
-export const SvgSearch: React.FC<{
+const StyledGradient = styled.div`
+  position: absolute;
+  bottom: 0;
+  z-index: 1;
+  background: linear-gradient(0deg, #fff 14%, ${rgba(255, 255, 255, 0)} 100%);
+  width: 100%;
+  height: 200px;
+`;
+
+const StyledCol = styled(Col).attrs({ forwardedAs: 'li' })``;
+const StyledRow = styled(Row).attrs({ forwardedAs: 'ul' })``;
+
+const StyledTokenContainer = styled(StyledRow)<{ isExpand: boolean }>`
+  position: relative;
+  overflow-y: hidden;
+  max-height: ${p => (p.isExpand ? `620px` : `inherit`)};
+`;
+
+interface ISvgNodeProps {
+  node: {
+    name: string;
+    relativeDirectory: string;
+    childGardenSvg: {
+      content: string;
+    };
+  };
+}
+
+interface ISvgSearchProps {
   searchEnabled?: boolean;
   data: {
-    edges: [
-      { node: { name: string; relativeDirectory: string; childGardenSvg: { content: string } } }
-    ];
+    edges: ISvgNodeProps[];
   };
-}> = ({ data, searchEnabled }) => {
-  const [inputValue, setInputValue] = useState('');
-  const [debouncedInputValue, setDebouncedInputValue] = useState(inputValue);
-
-  // eslint-disable-next-line react-hooks/exhaustive-deps
-  const updatedDebouncedInputValue = useCallback(
-    debounce((value: string) => {
-      setDebouncedInputValue(value);
-    }, 300),
-    []
-  );
-
-  useEffect(() => {
-    updatedDebouncedInputValue(inputValue);
-  }, [inputValue, updatedDebouncedInputValue]);
+}
 
-  const StyledCol = styled(Col).attrs({ forwardedAs: 'li' })``;
+const Icon = (edge: ISvgNodeProps) => {
+  return (
+    <StyledCol lg={3} md={4} xs={6}>
+      <StyledIconWrapper>
+        <StyledSvgWrapper
+          isAnswerBot={edge.node.name === 'answer-bot'}
+          // eslint-disable-next-line @typescript-eslint/naming-convention
+          dangerouslySetInnerHTML={{ __html: edge.node.childGardenSvg.content }}
+        />
+        <Code size="small" title={edge.node.name}>
+          {edge.node.name}
+        </Code>
+      </StyledIconWrapper>
+    </StyledCol>
+  );
+};
 
-  const StyledRow = styled(Row).attrs({ forwardedAs: 'ul' })``;
+export const SvgSearch: React.FC<ISvgSearchProps> = ({ data, searchEnabled }) => {
+  const [inputValue, setInputValue] = useState('');
+  const [collapsed, setCollapsed] = useState(true);
+  const debounceRef: any = useRef();
```

Comment:
Looks like this `ref` can be typed out to help the TS compiler:

```tsx
import type { DebouncedFunc } from 'lodash'; 

type ChangeHandler = (event: React.ChangeEvent<HTMLInputElement>) => void

const debounceRef = useRef<DebouncedFunc<ChangeHandler>>();

useEffect(() => {
  debounceRef.current = debounce((event) => {
    setInputValue(event.target.value);
  }, 300);

  return () => {
    if (debounceRef.current) {
      debounceRef.current.cancel();
    }
  };
}, []);


```

Reply:
Oh nice catch, forgot to set a type before pushing the changes. 😅